src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.Value.cs

```diff
@@ -35,49 +35,60 @@ private RuntimeOSPlatformInfo(string invokedPlatformCheckMethodName, string plat
             public bool Negated { get; }
 
             public IAbstractAnalysisValue GetNegatedValue()
-                => new RuntimeOSPlatformInfo(InvokedPlatformCheckMethodName, PlatformPropertyName, Version, !Negated);
+                => new RuntimeMethodValue(InvokedPlatformCheckMethodName, PlatformPropertyName, Version, !Negated);
 
             public static bool TryDecode(
                 IMethodSymbol invokedPlatformCheckMethod,
                 ImmutableArray<IArgumentOperation> arguments,
                 ValueContentAnalysisResult? valueContentAnalysisResult,
                 INamedTypeSymbol osPlatformType,
-                [NotNullWhen(returnValue: true)] out RuntimeOSPlatformInfo? info)
+                [NotNullWhen(returnValue: true)] out RuntimeMethodValue? info)
             {
-                if (!TryDecodeOSPlatform(arguments, osPlatformType, out var osPlatformProperty) ||
+                Debug.Assert(!arguments.IsEmpty);
+
+                if (arguments[0].Value is ILiteralOperation literal && literal.Type?.SpecialType == SpecialType.System_String)
```

Comment:
You are checking for a potential null type here but not before. Either it's not possible to have null and this is un-needed or the other part needs to be updated to handle null too.

Reply:
Not sure what you meant by "but not checking before" ...

Reply:
@Evangelink I am also not sure I understand this comment. Would you mind elaborating a bit more?